### ISSUE TITLE
Jupyter 4 compatibility

### ISSUE THREAD

**Issue by cameronrutherford** [jupyterlab-contrib/rise#29]
## Description

JupyterLab 4.0 came out recently and I wanted to use this with extensions like `jupyter_collaboration`, `jupyterlab-quatro`.

There is a migration guide https://jupyterlab.readthedocs.io/en/latest/extension/extension_migration.html#jupyterlab-3-x-to-4-x to update.

Unfortunately, quatro doesn't seem to support interactive presentations in the way rise does, so I would love this to be updated.

## Reproduce

I personally am using Apptainer on an HPC cluster, but a min reproducer is just installing the following container.

<details><summary>Dockerfile</summary>
<p>

```
FROM jupyter/minimal-notebook

RUN pip install --upgrade pip

RUN pip jupyter_collaboration \
                jupyterlab_rise
``` 

</p>
</details> 

## Expected behavior

I haven't attempted to update, but I imagine the first step is changing the pyproject.toml version https://github.com/jupyterlab-contrib/rise/blob/main/pyproject.toml#L2, and then figuring out what breaking API changes might have been made.

**Comment by nthiery** [comment]
I am currently testing jupyterlab-rise with jupyterlab 4.0, and it seems
to work smoothly (up to other issues already present with jupyterlab 3).

Presumably this was fixed by #39 and #42.

Does it work in your setting? If yes, maybe this issue can be closed.

**Comment by fcollonval** [comment]
Closing it is supported starting with [version 0.40.0](https://github.com/jupyterlab-contrib/rise/releases/tag/v0.40.0)
